pystac/extensions/pointcloud.py

```diff
@@ -0,0 +1,347 @@
+
+        If an Asset is supplied, sets the property on the Asset.
+        Otherwise sets the Item's value.
+        """
+        if asset is None:
+            self.item.properties['pc:count'] = count
+        else:
+            asset.properties['pc:count'] = count
+
+    @property
+    def type(self):
+        """Get or sets the pc:type prop on the Item
+
+        Returns:
+            str
+        """
+        return self.get_type()
+
+    @type.setter
+    def type(self, v):
+        self.set_type(v)
+
+    def get_type(self, asset=None):
+        """Gets an Item or an Asset type.
+
+        If an Asset is supplied and the Item property exists on the Asset,
+        returns the Asset's value. Otherwise returns the Item's value
+
+        Returns:
+            str
+        """
+        if asset is None or 'pc:type' not in asset.properties:
+            return self.item.properties.get('pc:type')
+        else:
+            return asset.properties.get('pc:type')
+
+    def set_type(self, type, asset=None):
+        """Set an Item or an Asset type.
+
+        If an Asset is supplied, sets the property on the Asset.
+        Otherwise sets the Item's value.
+        """
+        if asset is None:
+            self.item.properties['pc:type'] = type
+        else:
+            asset.properties['pc:type'] = type
+
+    @property
+    def encoding(self):
+        """Get or sets the content-encoding for the item.
+
+        The content-encoding is the underlying encoding format for the point cloud.
+        Examples may include: laszip, ascii, binary, etc.
+
+        Returns:
+            str
+        """
+        return self.get_encoding()
+
+    @encoding.setter
+    def encoding(self, v):
+        self.set_encoding(v)
+
+    def get_encoding(self, asset=None):
+        """Gets an Item or an Asset encoding.
+
+        If an Asset is supplied and the Item property exists on the Asset,
+        returns the Asset's value. Otherwise returns the Item's value
+
+        Returns:
+            str
+        """
+        if asset is None or 'pc:encoding' not in asset.properties:
+            return self.item.properties.get('pc:encoding')
+        else:
+            return asset.properties.get('pc:encoding')
+
+    def set_encoding(self, encoding, asset=None):
+        """Set an Item or an Asset encoding.
+
+        If an Asset is supplied, sets the property on the Asset.
+        Otherwise sets the Item's value.
+        """
+        if asset is None:
+            self.item.properties['pc:encoding'] = encoding
+        else:
+            asset.properties['pc:encoding'] = encoding
+
+    @property
+    def schemas(self):
+        """Get or sets a
+
+        The schemas represent the structure of the data attributes in the pointcloud,
+        and is represented as a sequential array of items that define the dimensions
+        and their types,
+
+        Returns:
+            List[dict]
+        """
+        return self.get_schemas()
```

Comment:
Would it make sense to define a `SchemaObject` explicitly as a Python object, or do you think it would be more convenient to keep it as a dict? The [label extension](https://github.com/stac-utils/pystac/blob/v0.5.2/pystac/extensions/label.py#L300) has an example of returning objects from extensions. I think for consistency it might be better to define an object, however I'll defer to your thoughts on usability as the first primary user of the extension.

If it does stay as `dict`, perhaps the docstring should include an example of a valid schema dict to make it easier for users to see how they should be constructed (similar to the 'statistics' example below)

Reply:
I like this. Makes sense i added PointcloudSchema and PointcloudStatistic as classes and tested get/set on each of their props too.